db/snapshot/backup_task.cc

```diff
@@ -118,37 +119,53 @@ future<> backup_task_impl::do_backup() {
 
 future<> backup_task_impl::process_snapshot_dir() {
     auto snapshot_dir_lister = directory_lister(_snapshot_dir, lister::dir_entry_types::of<directory_entry_type::regular>());
+    size_t num_sstable_comps = 0;
 
     try {
+        snap_log.debug("backup_task: listing {}", _snapshot_dir.native());
         while (auto component_ent = co_await snapshot_dir_lister.get()) {
             const auto& name = component_ent->name;
             auto file_path = _snapshot_dir / name;
             auto st = co_await file_stat(file_path.native());
             _total_progress.total += st.size;
-            _files.emplace_back(name);
+            try {
+                auto desc = sstables::parse_path(file_path, "", "");
+                const auto& gen = desc.generation;
+                _sstable_comps[gen].emplace_back(name);
+                ++num_sstable_comps;
+            } catch (const sstables::malformed_sstable_exception&) {
+                _files.emplace_back(name);
+            }
         }
+        snap_log.debug("backup_task: found {} SSTables consisting of {} component files, and {} non-sstable files",
+            _sstable_comps.size(), num_sstable_comps, _files.size());
     } catch (...) {
         _ex = std::current_exception();
+        snap_log.error("backup_task: listing {} failed: {}", _snapshot_dir.native(), _ex);
     }
 
     co_await snapshot_dir_lister.close();
     if (_ex) {
-        co_await coroutine::return_exception_ptr(_ex);
+        co_await coroutine::return_exception_ptr(std::move(_ex));
```

Comment:
Wait, what?! You "fixed" that back [here](https://github.com/scylladb/scylladb/pull/23241/commits/0daae622f85f69aec010c3c340f282dd27708a70#r2033108318) :exploding_head:

Reply:
oops, will undo in next version

Reply:
Undone in v8 ([63026c2](https://github.com/scylladb/scylladb/commit/63026c2151ada0845db2c8201b3367be799a130b))